frontend/tests/controllers/ContestList2Test.php

```diff
@@ -0,0 +1,63 @@
+<?php
+
+/**
+ * Description of Contest List v2
+ *
+ * @author Michael Serrato
+ */
+
+class ContestList2Test extends \OmegaUp\Test\ControllerTestCase {
+    public function testContestListDataForTypescript() {
+        // Get a contest
+        $contestData = \OmegaUp\Test\Factories\Contest::createContest(
+            new \OmegaUp\Test\Factories\ContestParams([
+                'requestsUserInformation' => 'optional',
+            ])
+        );
+
+        // Create user
+        ['identity' => $identity] = \OmegaUp\Test\Factories\User::createUser();
+
+        // Add user to our contest
+        \OmegaUp\Test\Factories\Contest::addUser(
+            $contestData,
+            $identity
+        );
+
+        // Log in user
+        $userLogin = self::login($identity);
+
+        $contestListDetails = \OmegaUp\Controllers\Contest::getContestListDetailsForTypeScript2(
+            new \OmegaUp\Request([
+                'auth_token' => $userLogin->auth_token,
+            ])
+        )['smartyProperties']['payload'];
+
+        $this->assertArrayHasKey('isLogged', $contestListDetails);
+        $this->assertIsBool($contestListDetails['isLogged']);
+
+        $this->assertArrayHasKey('contests', $contestListDetails);
+
+        $this->assertArrayHasKey('current', $contestListDetails['contests']);
+        $this->assertIsArray($contestListDetails['contests']['current']);
```

Comment:
aquí aseverar que algo es un arreglo no tiene mucho beneficio porque [Psalm](https://psalm.dev/) ya se encarga de que los tipos cuadren en PHP. en vez de eso, lo que hay que hacer es aseverar que se tienen _los valores_ correctos: en el comentario de arriba menciono que aquí lo mejor es checar que se contiene el concurso deseado:

```suggestion
        $this->assertEquals(
            array_map(fn ($contest) => $contest['alias'], $contestListDetails['contests']['current']),
            ['current_alias'],
        );
```

Reply:
Ok, vale.